unit_test/raw_image_test.py

```diff
@@ -0,0 +1,138 @@
+# Fluster - testing framework for decoders conformance
+# Copyright (C) 2021, Fluendo, S.A.
+#  Author: Manuel Jimeno Martínez <mjimeno@fluendo.com>, Fluendo, S.A.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Library General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+import unittest
+from typing import List
+from fluster.raw_image import RawImage, RawBufferType, ComparisonType
+
+
+class SyntheticImage:
+    """Generate a synthetic RGBA image"""
+
+    RED = [0xFF, 0, 0, 0xFF]
+    GREEN = [0, 0xFF, 0, 0xFF]
+    BLUE = [0, 0, 0xFF, 0xFF]
+
+    def __init__(
+        self,
+        first_band: List[int],
+        second_band: List[int],
+        third_band: List[int],
+        color_band_width: int,
+        num_rows: int,
+    ) -> None:
+        color_band_width = color_band_width // 3
+        _row = []
+        _row.extend(first_band * color_band_width)
+        _row.extend(second_band * color_band_width)
+        _row.extend(third_band * color_band_width)
+
+        self._image: bytearray = bytearray(_row * num_rows)
+
+    @property
+    def image(self) -> bytearray:
+        """Returns the image buffer"""
+        return self._image
+
+
+class TestRawImage(unittest.TestCase):
+    """Raw Image unit test"""
+
+    def test_frame_comparison_custom_rgba(self) -> None:
+        """Raw Image comparison of 2 rgba format"""
+        square_1 = SyntheticImage(
+            SyntheticImage.RED, SyntheticImage.GREEN, SyntheticImage.BLUE, 24, 24
+        )
+        square_2 = SyntheticImage(
+            SyntheticImage.RED, SyntheticImage.GREEN, SyntheticImage.BLUE, 24, 24
+        )
+        source = RawImage(square_1.image, "square_1", 24, 24, RawBufferType.RGBA)
+        target = RawImage(square_2.image, "square_2", 24, 24, RawBufferType.RGBA)
+        self.assertTrue(
```

Comment:
How can I use assertNotEqual or assertLess if they expect two values as parameters and I only return 1?

Reply:
You can do this: self.assertEqual(source.is_equal(target, RawBuffer.Type.RGBA), 0). It is clearer to the reader that this assertion checks the equality of an actual value to an expected, rather than that an expression is True.